doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,280 @@
+    --language-alias "$LANGUAGE_ALIAS"
+```
+
+## Usage
+
+The entry point of this framework is `AAF_RUN_QUERY_HANDLER` script. This script is simply a query loop which is responsible for executing the implemented algorithm.
+
+This script takes the necessary parameters to execute the desired algorithm in string json format. The json input includes two main parts:
+
+* `query_handler` : Details of the algorithm implemented by the user.
+* `temporary_output`:  Information where the temporary outputs (such as tables, bucketfs files, ...) of the query handler are kept. These temporary outputs will be removed after the execution of the QueryHandler.
+
+The following SQL statement shows how to call an AAF query handler:
+
+```sql
+EXECUTE SCRIPT AAF_RUN_QUERY_HANDLER('{
+    "query_handler": {
+        "factory_class": {
+            "module": "<CLASS_MODULE>",
+            "name": "<CLASS_NAME>"
+        },
+        "parameter": "<CLASS_PARAMETERS>",
+        "udf": {
+            "schema": "<UDF_DB_SCHEMA>",
+            "name": "<UDF_NAME>"
+        }
+    },
+    "temporary_output": {
+        "bucketfs_location": {
+            "connection_name": "<BUCKETFS_CONNECTION_NAME>",
+            "directory": "<BUCKETFS_DIRECTORY>"
+        },
+        "schema_name": "<TEMP_DB_SCHEMA>"
+    }
+}');
+```
+
+See [Implementation of Custom Algorithms](#implementation-of-custom-algorithms) for a complete example.
+
+### Parameters
+
+| Parameter                    | Required? | Description                                                                   |
+|------------------------------|-----------|-------------------------------------------------------------------------------|
+| `<CLASS_NAME>`               | yes       | Name of the query handler class                                               |
+| `<CLASS_MODULE>`             | yes       | Module name of the query handler class                                        |
+| `<CLASS_PARAMETERS>`         | yes       | Parameters of the query handler class encoded as string                       |
+| `<UDF_NAME>`                 | -         | Name of Python UDF script that contains the algorithm implemented by the user |
+| `<UDF_DB_SCHEMA>`            | -         | Schema name where the UDF script is deployed                                  |
+| `<BUCKETFS_CONNECTION_NAME>` | yes       | BucketFS connection name which is used to create temporary bucketfs files     |
+| `<BUCKETFS_DIRECTORY>`       | yes       | Directory in BucketFS for the temporary bucketfs files                        |
+| `<TEMP_DB_SCHEMA>`           | yes       | Database Schema for temporary database objects, e.g. tables                   |
+
+Please take care to provide a string value for `<CLASS_PARAMETERS>`.  Simple data types like `float`, `int`, `bool` will be converted to a String, while a Json object or an array is represented as a string value with an unusable reference, e.g.  `table: 0x14823bd38580`.
```

Comment:
We should that the json object problem will be fixed

Reply:
See next push.